### ISSUE TITLE
ElasticSearch Scroll API

### ISSUE THREAD

**Issue by adamtb705** [firebase/flashlight#77]
I have fully integrated Flashlight with my Firebase database and have several queries working perfectly already. I really love how nice everything works together. I have a particular use case in my iOS app where I need to request a huge number of items (sorting not necessary) to display on a map. Since this list could grow to greater than 100k results in the future I wanted to use ElasticSearch's scroll query string. I'm new to Node.js and am not sure exactly how to get Flashlight to do what I want. I am passing a moderately complex JSON query string as the query parameter and have set the index and type parameter to the correct index. This works perfect as one request, but is there any way I can get Flashlight to do a scroll pagination (i.e. _search?scroll=1m)? If not, do you have any recommendations on how to accomplish this?

Thanks so much for any help you can provide.
-Adam

**Comment by katowulf** [comment]
Hi Adam, you're looking for the `from=x` option in ES; it's covered under [ElasticSearch pagination](https://www.elastic.co/guide/en/elasticsearch/guide/current/pagination.html).

The placement in your call to Flashlight is shown in the README under [more on queries](https://www.elastic.co/guide/en/elasticsearch/guide/current/pagination.html).

**Comment by adamtb705** [comment]
Oh ok, great. I'm actually already doing this on another screen where the user actually scrolls, but I saw scroll and thought I would try to avoid the overhead. However, after reading some more of the documentation, I think the intended use of scrolling is not how I intended to use it. 

[https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html)

I think I could probably use scroll=1m in place of from=x as a parameter now that I see how the request works? Anyways, I'll stick with using from and size. 

Thanks again.
